File: backend/app/models/schema.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class ColumnSchema(BaseModel):
    name: str
    data_type: str
    column_type: str
    length: Optional[int] = None
    precision: Optional[int] = None
    scale: Optional[int] = None
    nullable: bool
    default_value: Optional[str] = None
    comment: str = ""
    primary_key: bool = False
    auto_increment: bool = False
    extra: str = ""
    column_key: str = ""
    ordinal_position: int


class TableSchema(BaseModel):
    datasource: str
    database: str
    table: str
    columns: List[ColumnSchema]
    table_comment: str = ""
    engine: str = ""
    charset: str = ""
    row_count: int = 0
    data_size: int = 0

File: frontend/src/types/schema.ts
export interface ColumnSchema {
  name: string
  data_type: string
  column_type: string
  length: number | null
  precision: number | null
  scale: number | null
  nullable: boolean
  default_value: string | null
  comment: string
  primary_key: boolean
  auto_increment: boolean
  extra: string
  column_key: string
  ordinal_position: number
}

export interface TableSchema {
  datasource: string
  database: string
  table: string
  columns: ColumnSchema[]
  table_comment: string
  engine: string
  charset: string
  row_count: number
  data_size: number
}
